move/move-ibc/sources/ibc.move

```diff
@@ -1230,120 +928,481 @@ module IBC::Core {
+            &store.commitments,
+            b"nextChannelSequence",
+            &bcs::to_bytes<u64>(&0)
+        );
+        let next_sequence = from_bcs::to_u64(*next_sequence_bytes);
+        let identifier = string::utf8(b"channel-");
+        string::append(&mut identifier, string_utils::to_string(&next_sequence));
+        let new_sequence = next_sequence + 1;
         table::upsert(
             &mut store.commitments,
-            IBCCommitment::next_sequence_send_commitment_key(port_id, channel_id),
-            bcs::to_bytes(&1)
+            b"nextChannelSequence",
+            bcs::to_bytes(&new_sequence)
         );
+        identifier
+    }
 
-        table::upsert(
-            &mut store.commitments,
-            IBCCommitment::next_sequence_recv_commitment_key(port_id, channel_id),
-            bcs::to_bytes(&1)
+    public fun ensure_connection_state(connection_id: String): ConnectionEnd acquires IBCStore {
+        let store = borrow_global<IBCStore>(get_vault_addr());
+        let connection = smart_table::borrow(
+            &store.connections,
+            connection_id,
         );
+        if (connection_end::state(connection) != CONN_STATE_OPEN) {
+            abort(E_INVALID_CONNECTION_STATE)
+        };
+        *connection
+    }
 
-        table::upsert(
-            &mut store.commitments,
-            IBCCommitment::next_sequence_ack_commitment_key(port_id, channel_id),
-            bcs::to_bytes(&1)
+    // TODO(aeryz): borrow instead of copy
+    public fun ensure_connection_feature(connection_hops: vector<String>, ordering: u8): (String, ConnectionEnd) acquires IBCStore {
+        if (vector::length(&connection_hops) != 1) {
+            abort(E_CONN_NOT_SINGLE_HOP)
+        };
+        let connection_id = *vector::borrow(&connection_hops, 0);
+        let connection = ensure_connection_state(connection_id);
+        if (vector::length(connection_end::versions(&connection)) != 1) {
+            abort(E_CONN_NOT_SINGLE_VERSION)
+        };
+        let version = *vector::borrow(connection_end::versions(&connection), 0);
+        if (!verify_supported_feature(&version, to_string(ordering))) {
+            abort(E_UNSUPPORTED_FEATURE)
+        };
+        (connection_id, connection)
+    }
+
+    // TODO: Here the implementation returns false if there is any uppercase character in the string
+    // eg: "portId" will return false but "port-0" will return true. Is this correct way to check?
+    public fun is_lowercase(s: &String): bool {
+        let bytes = bcs::to_bytes(s);
+        let i = 1; // First byte is the length of the string
+        while (i < vector::length(&bytes)) {
+            let byte = *vector::borrow(&bytes, i);
+            // if (byte < 0x61 || byte > 0x7A) { // ASCII values for 'a' and 'z'
+            //     return false;
+            // };
+            if (byte > 0x40 && byte < 0x5b) {
+                return false
+            };
+            i = i + 1;
+        };
+        true
+    }
```

Comment:
the reason we did this in the solidity contract is that we re-encode the port (contract address) as hex when indexing into the capabilities table, and if you create the channel with a checksummed address as the port id then it doesn't match and you can't actually send any packets on the channel. i'm not sure how this will work in move, but do keep it in mind